### ISSUE TITLE
reduced Hessian from solve_with_sens is returned scaled — expose the unscaled value (or the scaling factors to recover it)

### ISSUE THREAD

**Issue by jkitchin** [jkitchin/pounce#128]
## Summary

The reduced Hessian returned by `Problem.solve_with_sens(..., compute_reduced_hessian=True)` (and `Solver.reduced_hessian`) appears to be a **scaled** Hessian — returned in the solver's internal scaled coordinates rather than natural units. When used for the standard sIPOPT covariance recipe — pin the estimated parameters with equality constraints `P_k == θ̂_k`, take `H = B K⁻¹ Bᵀ` over the pin rows, and form `Cov(θ) = −inv(H)` — the result does **not** match the independently-validated statistical covariance. It is off by an overall factor that **tracks the number of finite elements (`nfe`)** in a collocation discretization, i.e. the constraint/element measure `h = T/nfe` is leaking into the reported reduced Hessian.

This has surprised us repeatedly across projects, which is the real motivation for filing: the value a covariance / identifiability workflow wants is the **unscaled** reduced Hessian, but what comes back is scaled, with no obvious, documented way to undo it. The core question:

> Should users expect the **unscaled** reduced Hessian (either by default, or via an explicit option), and if it stays scaled, can pounce **expose the scaling factors** so callers can recover the natural-units value reliably?

Right now each consumer is tempted to reverse-engineer the factor (≈ `nfe` here), which won't generalize to variable element sizes, multiple datasets, or other constraint scalings. Filing per discussion; happy to help with a fix/test.

## Expected

For a least-squares estimation NLP, `−B K⁻¹ Bᵀ` over the parameter-pin rows should equal the Fisher information (the reduced Hessian of the objective on the constraint manifold), so that `Cov(θ) = −inv(H)` matches the analytic/finite-difference covariance — **independent of the discretization** (`nfe`, element sizes).

## Actual

`−H` is proportional to the analytic FIM but scaled down by ≈ `nfe`, and the proportionality is not a clean scalar (the residual few-% shape difference is consistent with full-Lagrangian vs Gauss-Newton, which is fine — the concern is the **overall ≈ nfe scaling**).

Validated reference (A→B→C kinetics, concentration data, known-correct covariance cross-checked between an analytic Gauss-Newton FIM and a finite-difference Jacobian):

```
analytic FIM = inv(Cov):
  [[1.03e5, 2.42e5],
   [2.42e5, 4.25e6]]

pounce -H = -(B K⁻¹ Bᵀ), obj_scal=1, nfe=15:
  [[6.61e3, 1.59e4],
   [1.59e4, 2.82e5]]

elementwise FIM / (-H):
  nfe=10 : [10.08, 11.20]
  nfe=15 : [15.67, 15.08]
```

The factor tracks `nfe`. `obj_scal` only rescales `H` linearly (tried 0.5 / 1 / 2; none gives 1×). **`Problem.clear_problem_scaling()` has no effect** on the returned reduced Hessian — so this is not the gradient-based NLP auto-scaling, but the intrinsic scaling of how the collocation problem is posed (the DAE collocation constraint Jacobian carries `h`, which propagates through `K⁻¹`).

## Reproducer

Stack: `pounce` + `discopt` + `kipet2` (all `jkitchin/*`). The pin/extract path is:

```python
import numpy as np, pandas as pd
from scipy.integrate import odeint
from kipet2 import ReactionModel
from discopt._jax.nlp_evaluator import NLPEvaluator
from discopt._jax.parametric import extract_x_flat
from discopt.solvers.nlp_pounce import _IpoptCallbacks, _infer_constraint_bounds
import pounce

t = np.linspace(0, 6, 18); rng = np.random.default_rng(1)
C = odeint(lambda y,_,a,b:[-a*y[0], a*y[0]-b*y[1], b*y[1]], [1,0,0], t, args=(1.5,0.4))
df = pd.DataFrame({'A':C[:,0]+rng.normal(0,2e-3,len(t)),
                   'B':C[:,1]+rng.normal(0,2e-3,len(t)),
                   'C':C[:,2]+rng.normal(0,2e-3,len(t))}, index=t)
var = {'A':4e-6, 'B':4e-6, 'C':4e-6}

def run(nfe):
    r = ReactionModel('c')
    k1 = r.parameter('k1', value=1.0, bounds=(0.1,5)); k2 = r.parameter('k2', value=1.0, bounds=(0.05,3))
    A = r.component('A', value=1.0); B = r.component('B', value=0.0); Cc = r.component('C', value=0.0)
    r.add_ode('A', -k1*A); r.add_ode('B', k1*A-k2*B); r.add_ode('C', k2*B)
    r.add_data(data=df); r.set_discretization(nfe=nfe, ncp=3)
    r.run_opt(variances=var, covariance=True, cov_method='analytic')
    ro = r.results_dict['p_estimator']
    theta = {k: ro.P[k] for k in ('k1','k2')}
    FIM = np.linalg.inv(np.asarray(ro.covariance, float))   # validated reference

    b = r._make_builder(); m, dae, dmp = b.build_parameter_estimation_model(var)
    for nm in ('k1','k2'):
        m.subject_to(dmp[nm] == float(theta[nm]), name=f'pin_{nm}')   # pin constraints, appended last
    ev = NLPEvaluator(m); ncon = ev.n_constraints; pin = list(range(ncon-2, ncon))
    xf = np.asarray(extract_x_flat(b.solve_local(m), m))             # warm-start at the solution
    lb, ub = ev.variable_bounds; cl, cu = _infer_constraint_bounds(ev)
    prob = pounce.Problem(n=ev.n_variables, m=ev.n_constraints,
                          problem_obj=_IpoptCallbacks(ev), lb=lb, ub=ub, cl=cl, cu=cu)
    prob.add_option('print_level', 0)
    x, info = prob.solve_with_sens(xf, pin, compute_reduced_hessian=True, obj_scal=1.0)
    H = np.array(info['reduced_hessian'], float).reshape(2, 2, order='F')
    print(f'nfe={nfe}: FIM/(-H) diag = [{(FIM/(-H))[0,0]:.2f}, {(FIM/(-H))[1,1]:.2f}]')

for nfe in (10, 15, 20):
    run(nfe)
```

Observed: the diagonal of `FIM / (−H)` is ≈ `nfe` (10 → ~10–11, 15 → ~15).

## Questions / asks

1. Is the reduced Hessian from `solve_with_sens` / `Solver.reduced_hessian` intended to be returned in **scaled** coordinates? If so, the scaling convention should be documented.
2. Does pounce **expose the scaling vector(s)** it applied (variable/constraint scaling), so a caller can unscale to natural units — i.e. `Cov = −inv(D⁻¹ H D⁻¹)` or whatever the correct transform is? If not, could it?
3. Alternatively, could there be an option to return the **unscaled** reduced Hessian directly (the value a covariance/identifiability workflow actually wants)?

The goal is for `−inv(reduced_hessian)` to equal the statistical covariance reliably and problem-independently, so it can be used as a covariance/identifiability backend without a per-discretization fudge factor.
